pygame/math.py

```diff
@@ -0,0 +1,310 @@
+""" Math module """
+
+from numbers import Number
+
+
+swizzling = False
+
+
+def enable_swizzling():
+    """Globally enables swizzling for vectors.
+
+    Enables swizzling for all vectors until disable_swizzling() is called.
+    By default swizzling is disabled.
+    """
+    global swizzling
+    swizzling = True
+
+
+def disable_swizzling():
+    """Globally disables swizzling for vectors.
+
+    Disables swizzling for all vectors until enable_swizzling() is called.
+    By default swizzling is disabled.
+    """
+    global swizzling
+    swizzling = False
+
+
+class Vector2(object):
+    """A 2-Dimensional Vector."""
+
+    def __init__(self, *args):
+        if len(args) == 0:
+            self.x = 0.0
+            self.y = 0.0
+        elif len(args) == 1:
+            if isinstance(args[0], Vector2):
+                self.x = args[0].x
+                self.y = args[0].y
+            elif isinstance(args[0], basestring):
+                if args[0].startswith("<Vector2(") and args[0].endswith(")>"):
+                    tokens = args[0][9:-2].split(",")
+                    try:
+                        self.x = float(tokens[0])
+                        self.y = float(tokens[1])
+                    except ValueError:
+                        raise TypeError("Invalid string arguments - not floats ({}).".
+                                        format(args[0]))
+                else:
+                    raise TypeError("Invalid string argument - not like __repr__ ({}).".
+                                    format(args[0]))
+            elif len(args[0]) == 2:
+                self.x = args[0][0]
+                self.y = args[0][1]
+            else:
+                raise TypeError("Invalid argument length ({}).".format(len(args[0])))
+        else:
+            self.x = args[0]
+            self.y = args[1]
+
+    def __repr__(self):
+        return "<Vector2({}, {})>".format(self.x, self.y)
+
+    def __str__(self):
+        return "[{}, {}]".format(self.x, self.y)
+
+    def __len__(self):
+        return 2
+
+    def __getitem__(self, key):
+        """Provide indexed read access (vec[0] is x vec[1] is y)."""
+        if isinstance(key, slice):
```

Comment:
I wonder if we could implement this whole function as `return [self.x, self.y][key]`?